content/articles/eclat-algorithm-in-python/index.md

```diff
@@ -0,0 +1,154 @@
+
+Since we are all well with the intuitions, let's proceed and implement this algorithm in python.
+
+### Python Implementation of the Eclat Algorithm
+To have the best rules, we shall adopt the Apriori algorithm in our implementation. To get started, we need to import the necessary libraries for this session.
+
+```python
+# The first thing is to install this package
+!pip install apyori
+
+```
+### Data Preprocessing
+#### Importing the libraries
+The libraries we shall work with in this session are as follows;
+
+```python
+import numpy as np # to deal with numeric data
+import pandas as pd # to deal with dataframe
+
+```
+#### Dataset importation
+
+```python
+data = pd.read_csv('/content/drive/MyDrive/Market_Basket_Optimisation.csv', header = None) # read the dataset
+transact_list = [] # create an empty list to store the transactions
+for i in range(0, 7501):
+  transact_list.append([str(data.values[i,j]) for j in range(0, 20)]) # add the transactions to the list created above
+
+```
+### Eclat Algorithm
+Since we adopted the Apriori algorithm, we need to generate rules as follows;
+
+```python
+from apyori import apriori # import the apriori library
+rules = apriori(transactions = transact_list, min_support = 0.003, min_confidence = 0.2, min_lift = 3, min_length = 2, max_length = 2) # generate rules
+
+# list of results coming from the apriori model as a list
+rslt = list(rules)
+# Extracting support from the list of the results above
+
+```
+
+From the above list of rules, we need to extract all `support` and move a step ahead and organize them into a pandas data frame. Below is the code which will enable us to do so;
+
+### Organizing the above output in a pandas dataframe
+```python
+def inspect(rslt): # function to organize the output
+    left_handSide         = [tuple(result[2][0][0])[0] for result in rslt] # get the left hand side of the rules
+    right_handSide         = [tuple(result[2][0][1])[0] for result in rslt] # get the right hand side of the rules
+    supports    = [result[1] for result in rslt] # get the supports
+    return list(zip(left_handSide,right_handSide, supports)) # zip the above three lists together
+rslt_DataFrame = pd.DataFrame(inspect(rslt), columns = ['Product 1', 'Product 2', 'Support']) # create a pandas dataframe
+
+```
+Now that our output is organized into a pandas data frame, we can have a look at the first seven supports as follows;
+
+```python
+rslt_DataFrame.nlargest(n = 7, columns = 'Support') # printing the first 7 supports
+
+```
+![Arranged results](/engineering-education/eclat-algorithm-in-python/supports.png)
+ 
+From this output, we notice that the support of the first rule(herb & pepper with) is the highest, followed by the (whole wheat pasta with olive oil) ans so on. This table means that the first rule is the most important. Therefore, for the seller to maximize their sales and profit, they should follow the first rule.
```

Comment:
I'd like you to re-read this paragraph and correct the grammatical mistakes accordingly.